# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1522413870 -7200
#      Fri Mar 30 14:44:30 2018 +0200
# Branch stable
# Node ID 5fef743c34d76df4d4f5b7a995a37d50501b8ee0
# Parent  1cd3aeda7598a370dcf00540e4767237d209bc1d
build: Check for new help query member in qt 5.9 (bug #53006).

* documentation.cc (global_search): Use searchInput if related flag defined.

* acinclude.m4: Add test for QHelpSearchQueryWidget::searchInput(),
  which was introduced in 5.9 and which replaces query().

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -205,18 +205,24 @@ namespace octave
           }
 
         file.remove();
       }
   }
 
   void documentation::global_search (void)
   {
+#if defined (HAVE_QHELPSEARCHQUERYWIDGET_SEARCHINPUT)
+    QString queries
+      = m_help_engine->searchEngine ()->queryWidget ()->searchInput ();
+#else
     QList<QHelpSearchQuery> queries
       = m_help_engine->searchEngine ()->queryWidget ()->query ();
+#endif
+
     m_help_engine->searchEngine ()->search (queries);
   }
 
   void documentation::global_search_started (void)
   {
     qApp->setOverrideCursor(QCursor(Qt::WaitCursor));
   }
 
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -462,16 +462,48 @@ AC_DEFUN([OCTAVE_CHECK_FUNC_QHEADERVIEW_
     AC_LANG_POP(C++)
   ])
   if test $octave_cv_func_qheaderview_setsectionsmovable = yes; then
     AC_DEFINE(HAVE_QHEADERVIEW_SETSECTIONSMOVABLE, 1,
       [Define to 1 if you have the `QHeaderView::setSectionsMovable' member function.])
   fi
 ])
 dnl
+dnl Check whether the Qt QHelpSearchQueryWidget class has the searchInput
+dnl member function.  This function was introduced in Qt 5.9.
+dnl
+dnl FIXME: Delete this entirely when we drop support for Qt 5.8 or older.
+dnl
+AC_DEFUN([OCTAVE_CHECK_FUNC_QHELPSEARCHQUERYWIDGET_SEARCHINPUT], [
+  AC_CACHE_CHECK([for QHelpSearchQueryWidget::searchInput],
+    [octave_cv_func_qhelpsearchquerywidget_searchinput],
+    [AC_LANG_PUSH(C++)
+    ac_octave_save_CPPFLAGS="$CPPFLAGS"
+    ac_octave_save_CXXFLAGS="$CXXFLAGS"
+    CPPFLAGS="$QT_CPPFLAGS $CXXPICFLAG $CPPFLAGS"
+    CXXFLAGS="$CXXPICFLAG $CPPFLAGS"
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+        #include <QHelpSearchQueryWidget>
+        #include <QString>
+        ]], [[
+        QHelpSearchQueryWidget *query_widget = new QHelpSearchQueryWidget ();
+        QString search_input = query_widget->searchInput ();
+        ]])],
+      octave_cv_func_qhelpsearchquerywidget_searchinput=yes,
+      octave_cv_func_qhelpsearchquerywidget_searchinput=no)
+    CPPFLAGS="$ac_octave_save_CPPFLAGS"
+    CXXFLAGS="$ac_octave_save_CXXFLAGS"
+    AC_LANG_POP(C++)
+  ])
+  if test $octave_cv_func_qhelpsearchquerywidget_searchinput = yes; then
+    AC_DEFINE(HAVE_QHELPSEARCHQUERYWIDGET_SEARCHINPUT, 1,
+      [Define to 1 if you have the `QHelpSearchQueryWidget::searchInput' member function.])
+  fi
+])
+dnl
 dnl Check whether the Qt function qInstallMessageHandler is available.
 dnl This function was introduced in Qt 5.
 dnl
 dnl FIXME: Delete this entirely when we drop support for Qt 4.
 dnl
 AC_DEFUN([OCTAVE_CHECK_FUNC_QINSTALLMESSAGEHANDLER], [
   AC_CACHE_CHECK([for qInstallMessageHandler],
     [octave_cv_func_qinstallmessagehandler],
@@ -1756,17 +1788,17 @@ AC_DEFUN([OCTAVE_CHECK_QT_OPENGL_OFFSCRE
          #endif
          #if defined (HAVE_QOPENGLWIDGET)
          #  include <QOpenGLWidget>
          #  include <QOpenGLContext>
          #endif
          #if defined (HAVE_QOFFSCREENSURFACE)
          #  include <QOffscreenSurface>
          #endif
-         QOpenGLContext ctx;    
+         QOpenGLContext ctx;
          QOffscreenSurface surf;
        ]])],
        octave_cv_qt_opengl_os_ok=yes,
        octave_cv_qt_opengl_os_ok=no)
   ])
   CPPFLAGS="$ac_octave_save_CPPFLAGS"
   CXXFLAGS="$ac_octave_save_CXXFLAGS"
   AC_LANG_POP(C++)
@@ -1950,17 +1982,17 @@ AC_DEFUN([OCTAVE_CHECK_QT_VERSION], [AC_
     if test -z "$LRELEASE_QTVER"; then
       AC_CHECK_TOOLS(LRELEASE, [lrelease])
       if test -n "$LRELEASE" && test -n "$QTCHOOSER"; then
         LRELEASEFLAGS="-qt$qt_version"
       fi
     else
       LRELEASE="$LRELEASE_QTVER"
     fi
-    
+
     AC_CHECK_TOOLS(QCOLLECTIONGENERATOR_QTVER, [qcollectiongenerator-qt$qt_version])
     if test -z "$QCOLLECTIONGENERATOR_QTVER"; then
       AC_CHECK_TOOLS(QCOLLECTIONGENERATOR, [qcollectiongenerator])
       if test -n "$QCOLLECTIONGENERATOR" && test -n "$QTCHOOSER"; then
         QCOLLECTIONGENERATORFLAGS="-qt$qt_version"
       fi
     else
       QCOLLECTIONGENERATOR="$QCOLLECTIONGENERATOR_QTVER"
@@ -2059,16 +2091,17 @@ AC_DEFUN([OCTAVE_CHECK_QT_VERSION], [AC_
     ## We don't need to unset cache variables for any of the remaining
     ## tests if they fail because we have already decided that the Qt
     ## version that we are testing now will be the one used.
 
     OCTAVE_CHECK_FUNC_QGUIAPPLICATION_SETDESKTOPFILENAME
     OCTAVE_CHECK_FUNC_QHEADERVIEW_SETSECTIONRESIZEMODE
     OCTAVE_CHECK_FUNC_QHEADERVIEW_SETSECTIONSCLICKABLE
     OCTAVE_CHECK_FUNC_QHEADERVIEW_SETSECTIONSMOVABLE
+    OCTAVE_CHECK_FUNC_QHELPSEARCHQUERYWIDGET_SEARCHINPUT
     OCTAVE_CHECK_FUNC_QINSTALLMESSAGEHANDLER
     OCTAVE_CHECK_FUNC_QLINEEDIT_SETPLACEHOLDERTEXT
     OCTAVE_CHECK_FUNC_QMOUSEEVENT_LOCALPOS
     OCTAVE_CHECK_FUNC_QOBJECT_FINDCHILDREN_ACCEPTS_FINDCHILDOPTIONS
     OCTAVE_CHECK_FUNC_QTABWIDGET_SETMOVABLE
     OCTAVE_CHECK_FUNC_QTMESSAGEHANDLER_ACCEPTS_QMESSAGELOGCONTEXT
     OCTAVE_CHECK_MEMBER_QFONT_FORCE_INTEGER_METRICS
     OCTAVE_CHECK_MEMBER_QFONT_MONOSPACE
